src/Build.UnitTests/BackEnd/ResultsCache_Tests.cs

```diff
@@ -393,12 +394,49 @@ public void TestResultsCacheTranslation(object obj)
 
             var copy = new ResultsCache(TranslationHelpers.GetReadTranslator());
 
-            copy.ResultsDictionary.Keys.ToHashSet().SetEquals(resultsCache.ResultsDictionary.Keys.ToHashSet()).ShouldBeTrue();
+            CompareResultsCache(resultsCache, copy);
+        }
+
+        [Theory]
+        [InlineData(1, 1)] // Serialize version 0 and deserialize version 0 
+        [InlineData(1, 0)] // Serialize version 0 and deserialize latest version
+        public void TestResultsCacheTranslationAcrossVersions(int envValue1, int envValue2)
```

Comment:
Could you please clarify which scenario is failing during the deserialization of the dictionary?

Reply:
Currently, without this fix the scenario "Serialize version 0 and deserialize latest version" will fail. The cache is produced by older version of VS than consumer VS.

Reply:
Thank you! Could you please provide the example of actually failing when the env variable is not provided?

Reply:
when the environment variable is not provided - it is the previous test and it works. It is case "serialize latest version and deserialize latest version". The only case that does not is "serialize latest version and deserialize version 0". I have not added a negative test for that, because it is some random serialization error that will appear.